src/ffi.rs

```diff
@@ -225,3 +233,64 @@ pub unsafe extern "C" fn bdd_low(bdd: *mut BddPtr<'static>) -> *mut BddPtr<'stat
 pub unsafe extern "C" fn bdd_high(bdd: *mut BddPtr<'static>) -> *mut BddPtr<'static> {
     Box::into_raw(Box::new((*bdd).high()))
 }
+
+#[no_mangle]
+#[allow(clippy::missing_safety_doc)]
+pub unsafe extern "C" fn print_bdd(bdd: *mut BddPtr<'static>) -> *const c_char {
+    let s = std::ffi::CString::new((*bdd).print_bdd()).unwrap();
+    let p = s.as_ptr();
+    std::mem::forget(s);
+    p
+}
+
+#[no_mangle]
+#[allow(clippy::missing_safety_doc)]
+pub unsafe extern "C" fn bdd_wmc(
+    bdd: *mut BddPtr<'static>,
+    wmc: *mut WmcParams<RealSemiring>,
+) -> f64 {
+    DDNNFPtr::unsmoothed_wmc(&(*bdd), &(*wmc)).0
```

Comment:
@stites should this be unsmoothed? If the weights don't add up to `1.0`, this will give you an inaccurate WMC (I think this is why I didn't add this in the first pass).